### DISCUSSION TITLE
Order tailwind classes

### DISCUSSION THREAD

**Discussion by rubenesque-code** [ben-rogerson/twin.macro#673] [Q&A] [answered]
Hi there,

I'm wandering if there's currently a plugin, something akin to [prettier-plugin-tailwindcss](https://github.com/tailwindlabs/prettier-plugin-tailwindcss), or other method to order tailwind classes within a `tw` string?

Thanks.

**Comment by ben-rogerson** [comment, marked as answer]
I don't know of a prettier that will sort the tw import/prop styles. 
Best way forward would be to fork the tailwindcss plugin and enhance the javascript/typescript transformer.

**Reply by rubenesque-code** [reply]
Thanks! I'm not sure if I currently have the knowledge to do this though. Absolutely no worries if not and it might not be possible but is there a way you could point me in the right direction to do something like this (any guides, etc.)?

**Reply by ben-rogerson** [reply]
Couldn't point you towards a course-type resource but [ast explorer](https://astexplorer.net/#/gist/9771753532c4bcabe3cf77c666a901d9/d43e75fd6af40a951af844f3426df196bb653090) can really help you understand how to traverse the document:

[![image](https://user-images.githubusercontent.com/21288568/165281396-30f0d8b3-e55e-4c2a-8b67-b2092e8a60e8.png)](https://astexplorer.net/#/gist/9771753532c4bcabe3cf77c666a901d9/d43e75fd6af40a951af844f3426df196bb653090)

I'd also start messing around [in the code](https://github.com/tailwindlabs/prettier-plugin-tailwindcss/blob/v0.1.10/src/index.js#L294) and seeing how it currently works - you should get some ideas from there.

**Reply by rubenesque-code** [reply]
Thanks dude.
